### PR TITLE
feat(zk_toolbox): Update lint CI with `zk_toolbox`

### PR DESCRIPTION
**Pull request by matias-gonz**
## What ❔
Update CI with `zk_toolbox`

### CODE REVIEW

File: .github/workflows/ci-core-lint-reusable.yml [hunk 1]
```diff
@@ -19,23 +19,29 @@ jobs:
...
+          ci_run bash -c "git config --global --add safe.directory /usr/src/zksync"
+
       - name: Setup db
         run: |
-          ci_run zk
-          ci_run run_retried rustup show
-          ci_run zk db migrate
+          ci_run bash -c "mkdir -p /usr/src/zksync/volumes/reth/data"
+          ci_run zk_inception containers --observability false 
+          ci_run zk_supervisor database prepare
```
Comment (Deniallugo) [inline .github/workflows/ci-core-lint-reusable.yml:37]: why do you ned it?

Comment (matias-gonz) [reply .github/workflows/ci-core-lint-reusable.yml:38]: This is for the `zks database check`: https://github.com/matter-labs/zksync-era/pull/2694/files#diff-3c40f3b56452c5db2daa62cd14db43e11427c1d3392b537d9789458bbeaf2dc8L35-R48

File: .github/workflows/ci-zk-toolbox-reusable.yml [hunk 2]
```diff
@@ -115,6 +115,10 @@ jobs:
           --prover-db-url=postgres://postgres:notsecurepassword@postgres:5432 \
           --prover-db-name=zksync_prover_localhost_rollup
 
+      - name: Check Database
+        run: |
+          ci_run zk_supervisor database check-sqlx-data
+
```
Comment (Deniallugo) [inline .github/workflows/ci-zk-toolbox-reusable.yml:121]: why you moved it?

File: .github/workflows/ci-core-lint-reusable.yml [hunk 3]
```diff
@@ -19,23 +19,22 @@ jobs:
...
+      - name: Build
         run: |
-          ci_run zk
-          ci_run run_retried rustup show
-          ci_run zk db migrate
+          ci_run bash -c "cd zk_toolbox && cargo build --release"
+          cp ./zk_toolbox/target/release/zk_supervisor ./bin
+          cp ./zk_toolbox/target/release/zk_inception ./bin
+          ci_run bash -c "yarn install"
+          ci_run bash -c "git config --global --add safe.directory /usr/src/zksync"
```
Comment (Deniallugo) [inline .github/workflows/ci-core-lint-reusable.yml:31]: you don't need bash -c

File: .github/workflows/ci-core-lint-reusable.yml [hunk 4]
```diff
@@ -19,23 +19,22 @@ jobs:
...
 
-      - name: Setup db
+      - name: Build
         run: |
-          ci_run zk
-          ci_run run_retried rustup show
-          ci_run zk db migrate
+          ci_run bash -c "cd zk_toolbox && cargo build --release"
+          cp ./zk_toolbox/target/release/zk_supervisor ./bin
+          cp ./zk_toolbox/target/release/zk_inception ./bin
```
Comment (Deniallugo) [inline .github/workflows/ci-core-lint-reusable.yml:29]: why not `./bin/zkt`?

File: zk_toolbox/crates/zk_supervisor/src/commands/lint.rs [hunk 5]
```diff
@@ -19,6 +19,8 @@ pub struct LintArgs {
     pub check: bool,
     #[clap(long, short = 'e')]
     pub extensions: Vec<Extension>,
+    #[clap(long, default_missing_value = "true", num_args = 0..=1)]
+    pub contracts: Option<bool>,
```
Comment (Deniallugo) [inline zk_toolbox/crates/zk_supervisor/src/commands/lint.rs:23]: Why do you need it? why not `sol`?

Comment (matias-gonz) [reply zk_toolbox/crates/zk_supervisor/src/commands/lint.rs:23]: In the previous tool it was separated between `sol` files in the era repo and files in the contracts repo. The contract option lints files in the contract repo

File: .github/workflows/ci-zk-toolbox-reusable.yml (hunk 2 above)
Comment (matias-gonz) [reply .github/workflows/ci-zk-toolbox-reusable.yml:121]: So that we don't have to initialize the ecosystem every time we lint code

Comment (Deniallugo) [reply .github/workflows/ci-zk-toolbox-reusable.yml:121]: but, we need only run setup database. not the whole ecosystem

File: zk_toolbox/crates/zk_supervisor/src/commands/lint.rs (hunk 5 above)
Comment (Deniallugo) [reply zk_toolbox/crates/zk_supervisor/src/commands/lint.rs:23]: Them probably we need to rename extensions, and add it there?

File: .github/workflows/ci-zk-toolbox-reusable.yml (hunk 2 above)
Comment (matias-gonz) [reply .github/workflows/ci-zk-toolbox-reusable.yml:121]: `zk_supervisor database check-sqlx-data` needs a `secrets.yaml` which is provided by `ecosystem init`

File: zk_toolbox/crates/zk_supervisor/src/commands/lint.rs (hunk 5 above)
Comment (matias-gonz) [reply zk_toolbox/crates/zk_supervisor/src/commands/lint.rs:23]: Makes sense!

**Review by onyxet — APPROVED** [review]

**Pull request merged**